sw/device/lib/crypto/impl/integrity.c

```diff
@@ -0,0 +1,38 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/impl/integrity.h"
+
+#include "sw/device/lib/base/hardened.h"
+
+uint32_t integrity_unblinded_checksum(const crypto_unblinded_key_t *key) {
+  // TODO: decide on a checksum algorithm and implement integrity checks.
+  // TODO: maybe check the key length to make sure it's not pushing UINT32_MAX,
+  // as an overflow protection.
+  return 0;
+}
+
+uint32_t integrity_blinded_checksum(const crypto_blinded_key_t *key) {
+  // TODO: decide on a checksum algorithm and implement integrity checks.
+  // TODO: maybe check the key length to make sure it's not pushing UINT32_MAX,
+  // as an overflow protection.
+  return 0;
+}
+
+hardened_bool_t integrity_unblinded_key_check(
```

Comment:
nit^2: since this is `integrity_check`, how do you feel about naming these `key_{un,}blinded_integrity_check`?

Reply:
Sounds good -- although I might favor `{un,}blinded_key_integrity_check` so it matches the type names for keys better.

Reply:
Left these as-is for now, but if you have further thoughts I can add this small change to a follow-up PR.